services/provider/server/server.go

```diff
@@ -1801,3 +1877,671 @@ func getPeerCephFSID(ctx context.Context, cl client.Client, secretName, namespac
+		clientProfileName := util.CalculateMD5Hash(fmt.Sprintf("%s-ceph-rbd", clientName))
+		kubeResources = append(
+			kubeResources,
+			&csiopv1a1.ClientProfileMapping{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: consumer.Status.Client.Name,
+				},
+				Spec: csiopv1a1.ClientProfileMappingSpec{
+					Mappings: []csiopv1a1.MappingsSpec{
+						{
+							LocalClientProfile:  clientProfileName,
+							RemoteClientProfile: clientProfileName,
+							BlockPoolIdMapping:  blockPoolMapping,
+						},
+					},
+				},
+			},
+		)
+	}
+	return kubeResources, nil
+}
+
+func (s *OCSProviderServer) appendNoobaaKubeResources(
+	ctx context.Context,
+	kubeResources []client.Object,
+	consumer *ocsv1alpha1.StorageConsumer,
+) ([]client.Object, error) {
+	// Noobaa Configuration
+	// Fetch noobaa remote secret and management address and append to extResources
+	noobaaOperatorSecret := &v1.Secret{}
+	noobaaOperatorSecret.Name = fmt.Sprintf("noobaa-account-%s", consumer.Name)
+	noobaaOperatorSecret.Namespace = s.namespace
+
+	if err := s.client.Get(ctx, client.ObjectKeyFromObject(noobaaOperatorSecret), noobaaOperatorSecret); kerrors.IsNotFound(err) {
+		return kubeResources, nil
+	} else if err != nil {
+		return kubeResources, fmt.Errorf("failed to get %s secret. %v", noobaaOperatorSecret.Name, err)
+	}
+
+	authToken, ok := noobaaOperatorSecret.Data["auth_token"]
+	if !ok || len(authToken) == 0 {
+		return kubeResources, fmt.Errorf("auth_token not found in %s secret", noobaaOperatorSecret.Name)
+	}
+
+	noobaMgmtRoute := &routev1.Route{}
+	noobaMgmtRoute.Name = "noobaa-mgmt"
+	noobaMgmtRoute.Namespace = s.namespace
+
+	if err := s.client.Get(ctx, client.ObjectKeyFromObject(noobaMgmtRoute), noobaMgmtRoute); err != nil {
+		return kubeResources, fmt.Errorf("failed to get noobaa-mgmt route. %v", err)
+	}
+	if len(noobaMgmtRoute.Status.Ingress) == 0 {
+		return kubeResources, fmt.Errorf("no Ingress available in noobaa-mgmt route")
+	}
+
+	noobaaMgmtAddress := noobaMgmtRoute.Status.Ingress[0].Host
+	if noobaaMgmtAddress == "" {
+		return kubeResources, fmt.Errorf("no Host found in noobaa-mgmt route Ingress")
+	}
+
+	kubeResources = append(kubeResources,
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "noobaa-remote-join-secret",
+			},
+			Data: map[string][]byte{
+				"auth_token": authToken,
+				"mgmt_addr":  []byte(noobaaMgmtAddress),
+			},
+		},
+		&nbv1.NooBaa{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "noobaa-remote",
+			},
+			Spec: nbv1.NooBaaSpec{
+				JoinSecret: &v1.SecretReference{
+					Name: "noobaa-remote-join-secret",
+				},
+			},
+		},
+	)
+
+	return kubeResources, nil
+}
+
+func sanitizeKubeResource(obj client.Object) {
+
+	obj.SetUID("")
+	obj.SetResourceVersion("")
+	obj.SetGeneration(0)
+	obj.SetCreationTimestamp(metav1.Time{})
+	obj.SetDeletionTimestamp(nil)
+	obj.SetDeletionGracePeriodSeconds(nil)
+	obj.SetOwnerReferences(nil)
+	obj.SetFinalizers(nil)
+	obj.SetManagedFields(nil)
```

Comment:
Instead of trying to go over fields that need to be cleared, why not just create a zero object then copy the ones you need?